### PR TITLE
SPINEDEM-3794 -Improve Patient Access Mode Security

### PR DESCRIPTION
**Pull request by agelledi**
## Summary
* Routine Change
* Adds patient access mode security related tests

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.

### CODE REVIEW

**Review by guyflavell2-nhs — APPROVED** [review]

File: proxies/live/apiproxy/resources/jsc/RestrictRequest.js [hunk 1]
```diff
@@ -3,12 +3,14 @@ var request_pathsuffix = context.getVariable('proxy.pathsuffix');
 var request_path_nhs_number = request_pathsuffix.split("/")[2];
 var method_is_restricted = true
 var sync_wrapped = context.getVariable('request.header.x-sync-wrapped');
+var vot = context.getVariable('jwt.DecodeJWT.DecodeIdToken.claim.vot');
+var allowed_vots = ["P9.Cp.Cd","P9.Cm","P9.Cp.Ck"];
```
Comment (guyflavell2-nhs) [inline proxies/live/apiproxy/resources/jsc/RestrictRequest.js:8]: my js brain wants this to be a const.

Comment (richardseacome1-nhs) [reply proxies/live/apiproxy/resources/jsc/RestrictRequest.js:8]: that's what I originally tried, as per ecma 2015, but 'const' was not recognised on parsing and caused errors, which I was very surprised at

File: karate-tests/src/test/java/patients/patientAccess/getPatient.feature [hunk 2]
```diff
@@ -49,6 +49,7 @@ Feature: Patient Access (Retrieve)
       | patientType | nhsNumber   |
       | P0          | 9912003073  |
       | P5          | 9912003072  |
+      | P9.Cp       | 5900068196  |
```
Comment (claudiudumitrascu1-nhs) [inline karate-tests/src/test/java/patients/patientAccess/getPatient.feature:52]: [NHS Login test users](https://nhsd-confluence.digital.nhs.uk/pages/viewpage.action?spaceKey=APM&title=Testing+with+mock+auth#Testingwithmockauth-NHSLogintestusers) mentions a few P9s with no VOT, would be good to cover that scenario as well

File: proxies/live/apiproxy/resources/jsc/RestrictRequest.js [hunk 3]
```diff
@@ -3,12 +3,14 @@ var request_pathsuffix = context.getVariable('proxy.pathsuffix');
...
 
 if (request_pathsuffix.split("/")[1] == "_poll" && sync_wrapped == "true"){
     method_is_restricted = false
 }
 
-if (request_path_nhs_number == id_token_nhs_number){
+if ((request_path_nhs_number == id_token_nhs_number) && (allowed_vots.indexOf(vot) > -1)) {
```
Comment (claudiudumitrascu1-nhs) [inline proxies/live/apiproxy/resources/jsc/RestrictRequest.js:13]:
```suggestion
if ((request_path_nhs_number == id_token_nhs_number) && allowed_vots.includes(vot)) {
```
Might only make a functional difference when dealing with 0, undefined, NaN (so won't matter for our usage), mostly for readbility, for your consideration

**Review by claudiudumitrascu1-nhs — COMMENTED** [review]
A couple of comments for your consideration

File: proxies/live/apiproxy/resources/jsc/RestrictRequest.js (hunk 3 above)
Comment (richardseacome1-nhs) [reply proxies/live/apiproxy/resources/jsc/RestrictRequest.js:13]: I did start off with 'includes' but found it did not work very well and failed certain karate tests, so although includes more readable, I'd probably be inclined to stick with indexOf

**Review by claudiudumitrascu1-nhs — APPROVED** [review]

**Pull request merged**